source/core/ControlRef.cpp

```diff
@@ -39,6 +39,18 @@ class ControlRefNumManager : public RefNumManager {
  public:
     static ControlRefNumManager &ControlRefManager() { return _s_singleton; }
     static ControlRefNumType &RefNumStorage() { return _s_singleton.RefNumManager(); }
+    RefNum FindOrCreateRefNum(ControlRefInfo *info) {
+        ControlRefNumType::RefNumIterator ctlRefIter = _refStorage.Begin(), ctlRefEnd = _refStorage.End();
+        while (ctlRefIter != ctlRefEnd) {
+            if (ctlRefIter->second._refData.vi == info->vi && ctlRefIter->second._refData.controlTag->IsEqual(info->controlTag)) {
+                // Found an existing ref with the same data; return it, and dispose the new StringRef, because we now own it.
+                info->controlTag->Delete(info->controlTag);
```

Comment:
Sorry, I disagree -- controlTag is a StringRef but the static method is in the TypedArray base class; I think accessing the method via the variable type is more robust to changes in the class hierarchy.  What's confusing about it?

Reply:
The way TypedArrayCore and String are implemented now, I don't understand how accessing a static method using an object gives any robustness to changes in the class hierarchy. 

If the intent is to keep Delete() as static in TypedArrayCore and have another static/non-static method by the same name "Delete()" in the derived class, that would be a strong code smell.

OTOH, derived class itself can be passed as a template argument to base class and the base class method can in turn call the derived class static implementation. Though it would work, unless there is a strong reason not to make the method in base class virtual, it is not likely a good idea to do (and nothing of that sort is done here anyway currently).

In C# (not sure but likely Java too), it is a compiler error to access a static method using object.

Reply:
It's more robust because it uses the (declared, not virtual) type of the class to locate the static method to call.  If String was refactored to have a more efficient implementation which did not depend on TypedArrayCore but still conformed to the same API, this call would not need to change.
This is not C#.